File: backend/app/Models/ActivityNews.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class ActivityNews extends Model
{
    protected $table = 'activity_news';
    protected $primaryKey = 'activity_news_id';

    protected $fillable = [
        'title',
        'summary',
        'image_url',
        'release_date'
    ];

    public $timestamps = true;

    public function getImageUrlAttribute($value){
        return $value
            ? asset('storage/' . $value)
            : null;
    }
}

File: frontend/lib/models/activity_news.dart
class ActivityNews {
  final int id;
  final String title;
  final String? summary;
  final String? imageUrl;
  final String? releaseDate;

  ActivityNews({
    required this.id,
    required this.title,
    this.summary,
    this.imageUrl,
    this.releaseDate,
  });

  factory ActivityNews.fromJson(Map<String, dynamic> json) {
    return ActivityNews(
      id: int.tryParse(json['activity_news_id']?.toString() ?? '') ?? 0,
      title: json['title']?.toString() ?? '',
      summary: json['summary']?.toString(),
      imageUrl: json['image_url']?.toString(),
      releaseDate: json['release_date']?.toString(),
    );
  }
}
